types/utils.go

```diff
@@ -0,0 +1,37 @@
+package types
+
+import (
+	"github.com/tharsis/ethermint/crypto/ethsecp256k1"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/crypto/types/multisig"
+)
+
+// IsSupportedKey returns true if the pubkey type is supported by the chain
+// (i.e eth_secp256k1, amino multisig, ed25519).
+// NOTE: Nested multisigs are not supported.
+func IsSupportedKey(pubkey cryptotypes.PubKey) bool {
+	switch pubkey := pubkey.(type) {
+	case *ethsecp256k1.PubKey, *ed25519.PubKey:
+		return true
+	case multisig.PubKey:
+		if len(pubkey.GetPubKeys()) == 0 {
+			return false
+		}
+
+		for _, pk := range pubkey.GetPubKeys() {
+			switch pk.(type) {
+			case *ethsecp256k1.PubKey, *ed25519.PubKey:
```

Comment:
multisig can support ed25519 keys? when and how, thought this was only used for validators keys.

Reply:
yeah but it's a supported key, you can sign a tx with ed25519 on chain (it's always been supported)